File: testdata/rich/code/embeds.py
"""Module docstring: cctext embed fixture.

Contains the word def and a fence-like ``` sequence on purpose.
"""
import re

SQL = """
SELECT id, name
FROM users
WHERE active = 1
  AND name LIKE '%o''brien%'
"""


def greet(name):
    """Return a greeting for ``name``.

    >>> greet("x")
    'hi x'
    """
    return f"hi {name!r} {{literal braces}} {len(name) + 1}"


def find(text):
    '''Single-quoted docstring with "double" quotes inside.'''
    pat = r"^\d+\.\s+(\w+)\s*$"
    return re.match(pat, text)


ESCAPED = "a \"\"\" b"  # body contains an escaped triple quote
NESTED = "she said 'it''s \"fine\"'"
# def not_a_def(): """ this is a comment, not a docstring """
MIXED = 'single with "double" inside' + "double with 'single' inside"

File: testdata/rich/code/embeds.js
// cctext embed fixture: template literals, tagged templates, regex vs division.
const name = "world";
const plain = `hello ${name} and ${1 + 2} done`;

const q = sql`SELECT id, name FROM users WHERE id = ${userId} AND active = 1`;
const view = html`<div class="card"><span>${name}</span></div>`;

const re = /ab+c/gi;
const ratio = a / b / c;
const notRe = total / count; // division, not a regex literal

// TSX-style JSX snippet (this file is plain .js; a TSX grammar would embed it):
// <Foo bar="x">{y}</Foo>

const ticks = "string with `backticks` inside";
const alsoTicks = 'and \'single\' with ` too';

const doc = `
# Title

Some prose, then a fence:

\`\`\`python
def f():
    """docstring inside a template inside a fence"""
    return 1
\`\`\`

end of template`;

export { plain, q, view, re, ratio, notRe, ticks, alsoTicks, doc };
